# Déjame escuchar la música

# Contenido <a id='back'></a>

* [Introducción](#intro)
* [Etapa 1. Descripción de los datos](#data_review)
    * [Conclusiones](#data_review_conclusions)
* [Etapa 2. Preprocesamiento de datos](#data_preprocessing)
    * [2.1 Estilo del encabezado](#header_style)
    * [2.2 Valores ausentes](#missing_values)
    * [2.3 Duplicados](#duplicates)
    * [2.4 Conclusiones](#data_preprocessing_conclusions)
* [Etapa 3. Prueba de hipótesis](#hypothesis)
    * [3.1 Hipótesis 1: actividad de los usuarios y las usuarias en las dos ciudades](#activity)
* [Conclusiones](#end)

## Introducción <a id='intro'></a>
Como analista de datos, tu trabajo consiste en analizar datos para extraer información valiosa y tomar decisiones basadas en ellos. Esto implica diferentes etapas, como la descripción general de los datos, el preprocesamiento y la prueba de hipótesis.

Siempre que investigamos, necesitamos formular hipótesis que después podamos probar. A veces aceptamos estas hipótesis; otras veces, las rechazamos. Para tomar las decisiones correctas, una empresa debe ser capaz de entender si está haciendo las suposiciones correctas.

En este proyecto, compararás las preferencias musicales de las ciudades de Springfield y Shelbyville. Estudiarás datos reales de transmisión de música online para probar la hipótesis a continuación y comparar el comportamiento de los usuarios y las usuarias de estas dos ciudades.

### Objetivo:
Prueba la hipótesis:
1. La actividad de los usuarios y las usuarias difiere según el día de la semana y dependiendo de la ciudad.


### Etapas
Los datos del comportamiento del usuario se almacenan en el archivo `/datasets/music_project_en.csv`. No hay ninguna información sobre la calidad de los datos, así que necesitarás examinarlos antes de probar la hipótesis.

Primero, evaluarás la calidad de los datos y verás si los problemas son significativos. Entonces, durante el preprocesamiento de datos, tomarás en cuenta los problemas más críticos.

Tu proyecto consistirá en tres etapas:
 1. Descripción de los datos.
 2. Preprocesamiento de datos.
 3. Prueba de hipótesis.








[Volver a Contenidos](#back)

## Etapa 1. Descripción de los datos <a id='data_review'></a>

Abre los datos y examínalos.

Necesitarás `pandas`, así que impórtalo.

In [349]:
# Importar pandas
import pandas as pd

Lee el archivo `music_project_en.csv` de la carpeta `/datasets/` y guárdalo en la variable `df`:

In [350]:
# Leer el archivo y almacenarlo en df
music_data = pd.read_csv("music_proyect.csv")
music_df = pd.DataFrame(music_data)

Muestra las 10 primeras filas de la tabla:

In [351]:
# Obtener las 10 primeras filas de la tabla df
print(music_df.head(10))

     userID                        Track            artist   genre  \
0  FFB692EC            Kamigata To Boots  The Mass Missile    rock   
1  55204538  Delayed Because of Accident  Andreas Rönnberg    rock   
2    20EC38            Funiculì funiculà       Mario Lanza     pop   
3  A3DD03C9        Dragons in the Sunset        Fire + Ice    folk   
4  E2DC1FAE                  Soul People        Space Echo   dance   
5  842029A1                       Chains          Obladaet  rusrap   
6  4CB90AA5                         True      Roman Messer   dance   
7  F03E1C1F             Feeling This Way   Polina Griffith   dance   
8  8FA1D3BE                     L’estate       Julia Dalia  ruspop   
9  E772D5C0                    Pessimist               NaN   dance   

        City        time        Day  
0  Shelbyville  20:28:33  Wednesday  
1  Springfield  14:07:09     Friday  
2  Shelbyville  20:58:07  Wednesday  
3  Shelbyville  08:37:09     Monday  
4  Springfield  08:34:34     Monday  
5

Obtén la información general sobre la tabla con un comando. Conoces el método que muestra la información general que necesitamos.

In [352]:
# Obtener la información general sobre nuestros datos
music_df.info()
music_df.columns #Nombres de columnas para facilitar acceso: ['  userID', 'Track', 'artist', 'genre', '  City  ', 'time', 'Day']
print("Cantidad de duplicados: ", music_df.duplicated().sum())
music_df.isna().mean() #Artistas es lo que más falta o tiene valores nulos. 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65079 entries, 0 to 65078
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0     userID  65079 non-null  object
 1   Track     63736 non-null  object
 2   artist    57512 non-null  object
 3   genre     63881 non-null  object
 4     City    65079 non-null  object
 5   time      65079 non-null  object
 6   Day       65079 non-null  object
dtypes: object(7)
memory usage: 3.5+ MB
Cantidad de duplicados:  3826


  userID    0.000000
Track       0.020636
artist      0.116274
genre       0.018408
  City      0.000000
time        0.000000
Day         0.000000
dtype: float64

Estas son nuestras observaciones sobre la tabla. Contiene siete columnas. Almacenan los mismos tipos de datos: `object`.

Según la documentación:
- `' userID'`: identificador del usuario o la usuaria;
- `'Track'`: título de la canción;
- `'artist'`: nombre del artista;
- `'genre'`: género de la pista;
- `'City'`: ciudad del usuario o la usuaria;
- `'time'`: la hora exacta en la que se reprodujo la canción;
- `'Day'`: día de la semana.

Podemos ver tres problemas con el estilo en los encabezados de la tabla:
1. Algunos encabezados están en mayúsculas, otros en minúsculas.
2. Hay espacios en algunos encabezados.
3. `Detecta el tercer problema por tu cuenta y descríbelo aquí`.




El tercer problema a resolver es información nula en 3 columnas: Track, artist y genre (canción, artista y género). 
Por otra parte el tiempo es un objeto en lugar de un tipo de dato temporal o datetime64.  

### Escribe observaciones de tu parte. Estas son algunas de las preguntas que pueden ser útiles: <a id='data_review_conclusions'></a>

`1.   ¿Qué tipo de datos tenemos a nuestra disposición en las filas? ¿Y cómo podemos entender lo que almacenan las columnas?`

`2.   ¿Hay suficientes datos para proporcionar respuestas a nuestra hipótesis o necesitamos más información?`

`3.   ¿Notaste algún problema en los datos, como valores ausentes, duplicados o tipos de datos incorrectos?`

1. Tenemos filas sobre usuarios con ciudadanía descrita que escuchan canciones detalladas por artista y género a diferentes tiempos en los días de la semana. 
2. Considerando 65079 registros no soy experto pero me parece una muestra suficiente. No conozco la naturaleza del muestreo. 
3. Ausentes en tres columnas: canción (Track), artista (artist) y género (genre). 3826 duplicados. Tipo de datos incorrectos el de tiempo que debería ser datetime64. 

[Volver a Contenidos](#back)

## Etapa 2. Preprocesamiento de datos <a id='data_preprocessing'></a>

El objetivo aquí es preparar los datos para que sean analizados.
El primer paso es resolver cualquier problema con los encabezados. Luego podemos avanzar a los valores ausentes y duplicados. Empecemos.

Corrige el formato en los encabezados de la tabla.


### Estilo del encabezado <a id='header_style'></a>
Muestra los encabezados de la tabla (los nombres de las columnas):

In [353]:
# Muestra los nombres de las columnas
music_df.columns #['  userID', 'Track', 'artist', 'genre', '  City  ', 'time', 'Day']

Index(['  userID', 'Track', 'artist', 'genre', '  City  ', 'time', 'Day'], dtype='object')

Cambia los encabezados de la tabla de acuerdo con las reglas del buen estilo:
* Todos los caracteres deben ser minúsculas.
* Elimina los espacios.
* Si el nombre tiene varias palabras, utiliza snake_case.

Anteriormente, aprendiste acerca de la forma automática de cambiar el nombre de las columnas. Vamos a aplicarla ahora. Utiliza el bucle for para iterar sobre los nombres de las columnas y poner todos los caracteres en minúsculas. Cuando hayas terminado, vuelve a mostrar los encabezados de la tabla:

In [354]:
# Bucle en los encabezados poniendo todo en minúsculas
col_fixed = []
for col in music_df.columns: 
    col_fixed.append(col.lower()) #se puede usar todo junto .strip().lower()

print(col_fixed)#['  userid', 'track', 'artist', 'genre', '  city  ', 'time', 'day'] #aún le quedan espacios.


    

['  userid', 'track', 'artist', 'genre', '  city  ', 'time', 'day']


Ahora, utilizando el mismo método, elimina los espacios al principio y al final de los nombres de las columnas e imprime los nombres de las columnas nuevamente:

In [355]:
# Bucle en los encabezados eliminando los espacios
col_fixed2 = []
for col in col_fixed: 
    col_fixed2.append(col.strip())
    
print(col_fixed2)

['userid', 'track', 'artist', 'genre', 'city', 'time', 'day']


Necesitamos aplicar la regla de snake_case a la columna `userid`. Debe ser `user_id`. Cambia el nombre de esta columna y muestra los nombres de todas las columnas cuando hayas terminado.

In [356]:
# Cambiar el nombre de la columna "userid"
col_fixed2[0] = "user_id"
print(col_fixed2)

['user_id', 'track', 'artist', 'genre', 'city', 'time', 'day']


Comprueba el resultado. Muestra los encabezados una vez más:

In [357]:
# Comprobar el resultado: la lista de encabezados
music_df.columns = col_fixed2
music_df.columns

Index(['user_id', 'track', 'artist', 'genre', 'city', 'time', 'day'], dtype='object')

[Volver a Contenidos](#back)

### Valores ausentes <a id='missing_values'></a>
 Primero, encuentra el número de valores ausentes en la tabla. Debes utilizar dos métodos en una secuencia para obtener el número de valores ausentes.

In [358]:
# Calcular el número de valores ausentes
music_df.isna().sum() #1347 en track, 7567 en artist y 1198 en género. 

user_id       0
track      1343
artist     7567
genre      1198
city          0
time          0
day           0
dtype: int64

No todos los valores ausentes afectan a la investigación. Por ejemplo, los valores ausentes en `track` y `artist` no son cruciales. Simplemente puedes reemplazarlos con valores predeterminados como el string `'unknown'` (desconocido).

Pero los valores ausentes en `'genre'` pueden afectar la comparación entre las preferencias musicales de Springfield y Shelbyville. En la vida real, sería útil saber las razones por las cuales hay datos ausentes e intentar recuperarlos. Pero no tenemos esa oportunidad en este proyecto. Así que tendrás que:
* rellenar estos valores ausentes con un valor predeterminado;
* evaluar cuánto podrían afectar los valores ausentes a tus cómputos;

Reemplazar los valores ausentes en las columnas `'track'`, `'artist'` y `'genre'` con el string `'unknown'`. Como mostramos anteriormente en las lecciones, la mejor forma de hacerlo es crear una lista que almacene los nombres de las columnas donde se necesita el reemplazo. Luego, utiliza esta lista e itera sobre las columnas donde se necesita el reemplazo haciendo el propio reemplazo.

In [359]:
# Bucle en los encabezados reemplazando los valores ausentes con 'unknown'
col_a_cambiar = ["track", "artist", "genre"]
for column in col_a_cambiar: 
        music_df[column] = music_df[column].fillna("unknown") #inplace = True)

Ahora comprueba el resultado para asegurarte de que después del reemplazo no haya valores ausentes en el conjunto de datos. Para hacer esto, cuenta los valores ausentes nuevamente.

In [360]:
# Contar valores ausentes
music_df.isna().sum() #0 valores ausentes

user_id    0
track      0
artist     0
genre      0
city       0
time       0
day        0
dtype: int64

[Volver a Contenidos](#back)

### Duplicados <a id='duplicates'></a>
Encuentra el número de duplicados explícitos en la tabla. Una vez más, debes aplicar dos métodos en una secuencia para obtener la cantidad de duplicados explícitos.

In [361]:
# Contar duplicados explícitos
print(music_df.duplicated().sum()) #3826

3826


Ahora, elimina todos los duplicados. Para ello, llama al método que hace exactamente esto.

In [367]:
# Eliminar duplicados explícitos
music_df.drop_duplicates(inplace = True)

Comprobemos ahora si eliminamos con éxito todos los duplicados. Cuenta los duplicados explícitos una vez más para asegurarte de haberlos eliminado todos:

In [363]:
# Comprobar de nuevo si hay duplicados
print(music_df.duplicated().sum()) #0 duplicados


0


Ahora queremos deshacernos de los duplicados implícitos en la columna `genre`. Por ejemplo, el nombre de un género se puede escribir de varias formas. Dichos errores también pueden afectar al resultado.

Para hacerlo, primero mostremos una lista de nombres de género únicos, ordenados en orden alfabético. Para ello:
* Extrae la columna `genre` del DataFrame.
* Llama al método que devolverá todos los valores únicos en la columna extraída.


In [364]:
# Inspeccionar los nombres de géneros únicos
print(sorted(music_df["genre"].unique()))
#print(music_df["genre"].nunique()) #269 distintos géneros sin descartar typos. 


['acid', 'acoustic', 'action', 'adult', 'africa', 'afrikaans', 'alternative', 'ambient', 'americana', 'animated', 'anime', 'arabesk', 'arabic', 'arena', 'argentinetango', 'art', 'audiobook', 'avantgarde', 'axé', 'baile', 'balkan', 'beats', 'bigroom', 'black', 'bluegrass', 'blues', 'bollywood', 'bossa', 'brazilian', 'breakbeat', 'breaks', 'broadway', 'cantautori', 'cantopop', 'canzone', 'caribbean', 'caucasian', 'celtic', 'chamber', 'children', 'chill', 'chinese', 'choral', 'christian', 'christmas', 'classical', 'classicmetal', 'club', 'colombian', 'comedy', 'conjazz', 'contemporary', 'country', 'cuban', 'dance', 'dancehall', 'dancepop', 'dark', 'death', 'deep', 'deutschrock', 'deutschspr', 'dirty', 'disco', 'dnb', 'documentary', 'downbeat', 'downtempo', 'drum', 'dub', 'dubstep', 'eastern', 'easy', 'electronic', 'electropop', 'emo', 'entehno', 'epicmetal', 'estrada', 'ethnic', 'eurofolk', 'european', 'experimental', 'extrememetal', 'fado', 'film', 'fitness', 'flamenco', 'folk', 'folklor

Busca en la lista para encontrar duplicados implícitos del género `hiphop`. Estos pueden ser nombres escritos incorrectamente o nombres alternativos para el mismo género.

Verás los siguientes duplicados implícitos:
* `hip`
* `hop`
* `hip-hop`

Para deshacerte de ellos, crea una función llamada `replace_wrong_genres()` con dos parámetros:
* `wrong_genres=`: esta es una lista que contiene todos los valores que necesitas reemplazar.
* `correct_genre=`: este es un string que vas a utilizar como reemplazo.

Como resultado, la función debería corregir los nombres en la columna `'genre'` de la tabla `df`, es decir, remplazar cada valor de la lista `wrong_genres` por el valor en `correct_genre`.

Dentro del cuerpo de la función, utiliza un bucle `'for'` para iterar sobre la lista de géneros incorrectos, extrae la columna `'genre'` y aplica el método `replace` para hacer correcciones.

In [ ]:
# Función para reemplazar duplicados implícitos
def replace_wrong_genres(wrong_genres, correct_genre, df=music_df):
    for wrong_genre in wrong_genres:
            df["genre"] = df["genre"].replace(wrong_genre, correct_genre) #inplace = True) 
            #usar inplace funciona pero genera un mensaje indeseado en la terminal


Ahora, llama a `replace_wrong_genres()` y pásale tales argumentos para que retire los duplicados implícitos (`hip`, `hop` y `hip-hop`) y los reemplace por `hiphop`:

In [370]:
# Eliminar duplicados implícitos
wrong_genres = ["hip", "hop", "hip-hop"]
correct_genre = "hiphop"
replace_wrong_genres(wrong_genres, correct_genre)

Asegúrate de que los nombres duplicados han sido eliminados. Muestra la lista de valores únicos de la columna `'genre'` una vez más:

In [ ]:
# Comprobación de duplicados implícitos
print(music_df["genre"].unique())
music_df["genre"].nunique() #266 

['rock' 'pop' 'folk' 'dance' 'rusrap' 'ruspop' 'world' 'electronic'
 'unknown' 'alternative' 'children' 'rnb' 'hiphop' 'jazz' 'postrock'
 'latin' 'classical' 'metal' 'reggae' 'triphop' 'blues' 'instrumental'
 'rusrock' 'dnb' 'türk' 'post' 'country' 'psychedelic' 'conjazz' 'indie'
 'posthardcore' 'local' 'avantgarde' 'punk' 'videogame' 'techno' 'house'
 'christmas' 'melodic' 'caucasian' 'reggaeton' 'soundtrack' 'singer' 'ska'
 'salsa' 'ambient' 'film' 'western' 'rap' 'beats' "hard'n'heavy"
 'progmetal' 'minimal' 'tropical' 'contemporary' 'new' 'soul' 'holiday'
 'german' 'jpop' 'spiritual' 'urban' 'gospel' 'nujazz' 'folkmetal'
 'trance' 'miscellaneous' 'anime' 'hardcore' 'progressive' 'korean'
 'numetal' 'vocal' 'estrada' 'tango' 'loungeelectronic' 'classicmetal'
 'dubstep' 'club' 'deep' 'southern' 'black' 'folkrock' 'fitness' 'french'
 'disco' 'religious' 'drum' 'extrememetal' 'türkçe' 'experimental' 'easy'
 'metalcore' 'modern' 'argentinetango' 'old' 'swing' 'breaks' 'eurofolk'
 'stone

266

[Volver a Contenidos](#back)

### Tus observaciones <a id='data_preprocessing_conclusions'></a>

`Describe brevemente lo que has notado al analizar duplicados, cómo abordaste sus eliminaciones y qué resultados obtuviste.`

Al analizar duplicados observé que había 3826 y evidentemente fueron retirados con drop_duplicates, no recordaba el inplace y me causó problemas. De los duplicados no explícitos retiré 3 typos sin embargo quedó inconclusa la revisión de typos para saber que si hay otros generos redundantes y reemplazables. 
Abordé la eliminación iterando sobre cada valor en la lista de los géneros erróneos donde cada valor si se encuentra en la columna género del marco de datos será reemplazada por el valor corregido (str). Me quedo inconforme de la revisión de duplicados no implícitos y no vi los 266, supongo es una tarea artesánica o quizá sabiendo usar regex se pueda facilitar. 

[Volver a Contenidos](#back)

## Etapa 3. Prueba de hipótesis <a id='hypothesis'></a>

### Hipótesis: comparar el comportamiento del usuario o la usuaria en las dos ciudades <a id='activity'></a>

La hipótesis afirma que existen diferencias en la forma en que los usuarios y las usuarias de Springfield y Shelbyville consumen música. Para comprobar esto, usa los datos de tres días de la semana: lunes, miércoles y viernes.

* Agrupa a los usuarios y las usuarias por ciudad.
* Compara el número de canciones que cada grupo reprodujo el lunes, el miércoles y el viernes.


Realiza cada cálculo por separado.

El primer paso es evaluar la actividad del usuario en cada ciudad. Recuerda las etapas dividir-aplicar-combinar de las que hablamos anteriormente en la lección. Tu objetivo ahora es agrupar los datos por ciudad, aplicar el método apropiado para contar durante la etapa de aplicación y luego encontrar la cantidad de canciones reproducidas en cada grupo especificando la columna para obtener el recuento.

A continuación se muestra un ejemplo de cómo debería verse el resultado final:
`df.groupby(by='....')['column'].method()`Realiza cada cálculo por separado.

Para evaluar la actividad de los usuarios y las usuarias en cada ciudad, agrupa los datos por ciudad y encuentra la cantidad de canciones reproducidas en cada grupo.



In [ ]:
# Contar las canciones reproducidas en cada ciudad
#music_df.info()
#music_df.columns #'user_id', 'track', 'artist', 'genre', 'city', 'time', 'day'
df_canciones_por_ciudad = music_df.groupby("city")["track"].count()
print(df_canciones_por_ciudad) #Shelbyville: 18512, Springfield: 42741
#print(df_canciones_por_ciudad.Springfield/df_canciones_por_ciudad.Shelbyville)

city
Shelbyville    18512
Springfield    42741
Name: track, dtype: int64
2.3088267070008643


`Comenta tus observaciones aquí`

Más del doble de canciones escuchadas en Springfield. Parece ser una ciudad con más inclinación a escuchar música. Primer indicio interesante sobre la hipótesis. Por ciudad se escucha más música en Springfield. Ahora falta ver por día.

Ahora agrupemos los datos por día de la semana y encontremos el número de canciones reproducidas el lunes, miércoles y viernes. Utiliza el mismo método que antes, pero ahora necesitamos una agrupación diferente.


In [391]:
# Calcular las canciones reproducidas en cada uno de los tres días
df_canciones_por_día = music_df.groupby("day")["track"].count()
print(df_canciones_por_día) #Lunes: 21354, Miercoles: 18059, Viernes: 21840.
print((df_canciones_por_día["Friday"]/df_canciones_por_día["Monday"])*100) #Al rededor de un 2.3% más.
promedio_viernes_y_lunes = sum([df_canciones_por_día["Friday"], df_canciones_por_día["Monday"]])/2
print((df_canciones_por_día["Wednesday"]/promedio_viernes_y_lunes)*100)

day
Friday       21840
Monday       21354
Wednesday    18059
Name: track, dtype: int64
102.27592020230402
83.61809510580173


`Comenta tus observaciones aquí`

Los viernes se escucha un poco más de musica (21840) seguido por lunes (21354). Tan solo un 2.3% más. 
Los miércoles (18059) se escucha 14.4% menos música que el promedio de lunes y viernes. Será significativamente menos? Qué pasará con la parte de que cambia el comportamiento por día en la hipótesis? 

Ya sabes cómo contar entradas agrupándolas por ciudad o día. Ahora necesitas escribir una función que pueda contar entradas según ambos criterios simultáneamente.

Crea la función `number_tracks()` para calcular el número de canciones reproducidas en un determinado día **y** ciudad. La función debe aceptar dos parámetros:

- `day`: un día de la semana para filtrar. Por ejemplo, `'Monday'` (lunes).
- `city`: una ciudad para filtrar. Por ejemplo, `'Springfield'`.

Dentro de la función, aplicarás un filtrado consecutivo con indexación lógica.

Primero filtra los datos por día y luego filtra la tabla resultante por ciudad.

Después de filtrar los datos por dos criterios, cuenta el número de valores de la columna 'user_id' en la tabla resultante. Este recuento representa el número de entradas que estás buscando. Guarda el resultado en una nueva variable y devuélvelo desde la función.

In [ ]:
# Declara la función number_tracks() con dos parámetros: day= y city=.
def number_tracks(day, city, df=music_df): 
    por_día_df = df[df["day"] == day]
    por_día_y_ciudad_df = por_día_df[por_día_df["city"]==city]
    por_día_y_ciudad_df["user_id"].count()
    número_user_id = por_día_y_ciudad_df["user_id"].count()
    return número_user_id
 
    # Almacena las filas del DataFrame donde el valor en la columna 'day' es igual al parámetro day=
    # Filtra las filas donde el valor en la columna 'city' es igual al parámetro city=
    # Extrae la columna 'user_id' de la tabla filtrada y aplica el método count()
    # Devolve el número de valores de la columna 'user_id'

Llama a `number_tracks()` seis veces, cambiando los valores de los parámetros para que recuperes los datos de ambas ciudades para cada uno de los tres días.

In [ ]:
# El número de canciones reproducidas en Springfield el lunes
canciones_springfield_lunes = number_tracks("Monday", "Springfield")
print(canciones_springfield_lunes)

15740


In [ ]:
# El número de canciones reproducidas en Shelbyville el lunes
canciones_shelbyville_lunes = number_tracks("Monday", "Shelbyville")
print(canciones_shelbyville_lunes)

5614


In [400]:
# El número de canciones reproducidas en Springfield el miércoles
canciones_springfield_miércoles = number_tracks("Wednesday", "Springfield")
print(canciones_springfield_miércoles)

11056


In [401]:
# El número de canciones reproducidas en Shelbyville el miércoles
canciones_shelbyville_miércoles = number_tracks("Wednesday", "Shelbyville")
print(canciones_shelbyville_miércoles)

7003


In [ ]:
# El número de canciones reproducidas en Springfield el viernes
canciones_springfield_viernes = number_tracks("Friday", "Springfield")
print(canciones_springfield_miércoles)

11056


In [ ]:
# El número de canciones reproducidas en Shelbyville el viernes
canciones_shelbyville_miércoles = number_tracks("Friday", "Shelbyville")
print(canciones_shelbyville_miércoles)

11056


**Conclusiones**

`Comenta si la hipótesis es correcta o se debe rechazar. Explica tu razonamiento.`

[Volver a Contenidos](#back)

# Conclusiones <a id='end'></a>

`Resume aquí tus conclusiones sobre la hipótesis.`

La hipótesis sobre que el comportamiento de escucha de canciones varía por según el día y por ciudad es aceptada/correcta. Por ciudad se escucha poco más de 2 (230%; 2.3) veces más canciones en springfield que en Shelbyvile y varían en todos los días naturalmente habiendo más en springfield. Se podrían ver proporciones para evaluaciones de hipótesis más rigurosas e informativas.

### Nota
En proyectos de investigación reales, la prueba de hipótesis estadística es más precisa y cuantitativa. También ten en cuenta que no siempre se pueden sacar conclusiones sobre una ciudad entera a partir de datos de una sola fuente.

Aprenderás más sobre la prueba de hipótesis en el sprint de análisis estadístico de datos.

[Volver a Contenidos](#back)